### ISSUE TITLE
Strictness of strict_date_optional_time date format in 7.0.0

### ISSUE THREAD

**Issue by russcam** [elastic/elasticsearch#41927]
**Elasticsearch version** (`bin/elasticsearch --version`): 7.0.0

**Plugins installed**: []

**JVM version** (`java -version`):

java version "1.8.0_201"
Java(TM) SE Runtime Environment (build 1.8.0_201-b09)
Java HotSpot(TM) 64-Bit Server VM (build 25.201-b09, mixed mode)

**OS version** (`uname -a` if on a Unix-like system):

```ps
Get-CimInstance Win32_OperatingSystem | Select-Object Caption, Version, ServicePackMajorVersion, OSArchitecture | Format-List
```
Caption                 : Microsoft Windows 10 Pro
Version                 : 10.0.17763
ServicePackMajorVersion : 0
OSArchitecture          : 64-bit

**Description of the problem including expected versus actual behavior**:

The strictness of the default formatting applied to a `date` property between 6.7.2 and 7.0.0 causes requests that succeeded against 6.7.2 to fail against 7.0.0.

I think this may possibly be related to `strict_date_optional_time` requiring 3 places for sub second values in 7.0.0.

https://github.com/elastic/elasticsearch/blob/37758bb58306a9a319ec46e42aa8aead7a3acc5d/server/src/main/java/org/elasticsearch/common/time/DateFormatters.java#L113

**Steps to reproduce**:

Please include a *minimal* but *complete* recreation of the problem, including
(e.g.) index creation, mappings, settings, query etc.  The easier you make for
us to reproduce it, the more likely that somebody will take the time to look at it.

 1. Start Elasticsearch 6.7.2
 2. Execute the following requests

```json
PUT http://localhost:9200/default_index?pretty=true 
{
  "mappings": {
    "_doc": {
      "properties": {
        "createdDate": {
          "type": "date"
        }
      }
    }
  }
}

POST http://localhost:9200/default_index/_doc?pretty=true&refresh=wait_for 
{
  "createdDate": "2019-05-07T12:00:00.01"
}
```
 3. See that 6.7.2 happily accepts the date string `"2019-05-07T12:00:00.01"`
 4. Stop Elasticsearch 6.7.2 and start Elasticsearch 7.0.0 
 5. Execute the following requests

```json
PUT http://localhost:9200/default_index?pretty=true 
{"mappings":{"properties":{"createdDate":{"type":"date"}}}}

POST http://localhost:9200/default_index/_doc?pretty=true&refresh=wait_for 
{"createdDate":"2019-05-07T12:00:00.01"}
```
  
  See that indexing the document with date string fails "2019-05-07T12:00:00.01"

```json
Status: 400
{
  "error" : {
    "root_cause" : [
      {
        "type" : "mapper_parsing_exception",
        "reason" : "failed to parse field [createdDate] of type [date] in document with id '0LUdlWoB3J8uIm0O7hI4'"
      }
    ],
    "type" : "mapper_parsing_exception",
    "reason" : "failed to parse field [createdDate] of type [date] in document with id '0LUdlWoB3J8uIm0O7hI4'",
    "caused_by" : {
      "type" : "illegal_argument_exception",
      "reason" : "failed to parse date field [2019-05-07T12:00:00.01] with format [strict_date_optional_time||epoch_millis]",
      "caused_by" : {
        "type" : "date_time_parse_exception",
        "reason" : "Failed to parse with all enclosed parsers"
      }
    }
  },
  "status" : 400
}
```

**Comment by jasontedor** [comment]
I think this was recently addressed by #41871.

**Comment by jasontedor** [comment]
Duplicates #41633
